### PR TITLE
Dependent typing

### PR DESCRIPTION
**Pull request by RobbieMcKinstry**
Adds support for simple dependent types.

### CODE REVIEW

**Comment by RobbieMcKinstry** [comment]
I'm fairly sure that this only works if the dependent type is in the return value. If it's a parameter, I believe it will fail.

File: tools/src/wyvern/target/corewyvernIL/support/InvocationExprGenerator.java [hunk 1]
```diff
@@ -28,7 +29,7 @@ public InvocationExprGenerator(Expression receiver, String operationName, GenCon
 	
 	@Override
 	public Expression genExpr() {
-		if (declType instanceof ValDeclType || declType instanceof VarDeclType) {
+		if (declType instanceof ValDeclType || declType instanceof VarDeclType || declType instanceof ConcreteTypeMember) {
```
Comment (JonathanAldrich) [inline tools/src/wyvern/target/corewyvernIL/support/InvocationExprGenerator.java]: Actually it's not legal to create a FieldGet when the field is a ConcreteTypeMember; the result is not a legal expression and should not (for example) typecheck or do anything meaningful at run time.

**Comment by JonathanAldrich** [comment]
I made a comment on one change that doesn't look quite right--but decided to merge the commit as the rest was in good shape.

**Comment by JonathanAldrich** [comment]
I just backed out the one change in a separate commit; tests pass so it should be OK.

**Comment by RobbieMcKinstry** [comment]
Okay, sounds good. I don't think I understand why it's not a logical to have a FieldGet on a Concrete type member, so we'll talk about that when we meet in an hour.

**Pull request merged**